packages/webapp/src/containers/CustomSignUp/saga.js

```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright (C) 2007 Free Software Foundation, Inc. <https://fsf.org/>
+ *  This file (saga.js) is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { createAction } from '@reduxjs/toolkit';
+import { put, takeLatest, call } from 'redux-saga/effects';
+import { loginUrl as url } from '../../apiConfig';
+import history from '../../history';
+import { saveUserEmailSuccess, saveUserNameSuccess } from './signUpSlice';
+
+const axios = require('axios');
+const loginUrl = (email) => `${url}/user/${email}`;
+
+export const customSignUp = createAction(`customSignUpSaga`);
+
+export function* customSignUpSaga({ payload: email }) {
+  try {
+    const result = yield call(axios.get, loginUrl(email));
+    if (result.data.user != null) {
```

Comment:
I'm not quite sure what this check is for. What is the logic of having `saveUserEmailSuccess` be called two times here?

Reply:
Also, always try to use `!==`  instead of `!=` Or, Strict isntead of Abstract equality [the why](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness). Unless there's a specific reason to avoid it.

Reply:
I was saving it twice because it hits two different cases -- if a user is a new user, `result.data.user` will be null. I need to save their email to retrieve it in create user account. If they are an existing user, I also want to save it so that after a user inputs their password in the password page, the email can be passed to the `authenticateUser` -- but I just realized this endpoint does not even require the email. So can remove the latter.